File: src/main/java/com/linebot/entity/Notice.java
package com.linebot.entity;

import lombok.*;

import javax.persistence.*;
import java.util.Date;

@Builder
@Data
@Entity
@Table(name = "notice")
@NoArgsConstructor
@AllArgsConstructor
public class Notice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;
    @Column(name = "user_id")
    private String userId;
    @Column(name = "year")
    private int year;
    @Column(name = "month")
    private int month;
    @Column(name = "day")
    private int day;
    @Column(name = "hour")
    private int hour;
    @Column(name = "minute")
    private int minute;
    @Column(name = "day_of_week")
    private int dayOfWeek;
    @Column(name = "type")
    private int type;
    @Column(name = "deleted")
    private String deleted;
    @Column(name = "created_at")
    private Date createdAt;

    @AllArgsConstructor
    public enum Type {
        TRAIN_DELAY(1),
        CRON(2);
        @Getter
        private int code;
    }
}

File: src/main/kotlin/com/linebot/entity/Notice.kt
package com.linebot.entity

import java.util.Date
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "notice")
data class Notice(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Int? = null,

    @Column(name = "user_id")
    var userId: String? = null,

    @Column(name = "year")
    var year: Int = 0,

    @Column(name = "month")
    var month: Int = 0,

    @Column(name = "day")
    var day: Int = 0,

    @Column(name = "hour")
    var hour: Int = 0,

    @Column(name = "minute")
    var minute: Int = 0,

    @Column(name = "day_of_week")
    var dayOfWeek: Int = 0,

    @Column(name = "type")
    var type: Int = 0,

    @Column(name = "deleted")
    var deleted: String? = null,

    @Column(name = "created_at")
    var createdAt: Date? = null
) {
    enum class Type(
        val code: Int
    ) {
        NORMAL(1),
        SPECIAL(2);
    }
}
